### PR TITLE
Standalone package based installation

### LINKED ISSUE #29: Add to documentation Standalone / Package based installation

### PR DESCRIPTION
**Pull request by cpprian**

### CODE REVIEW

File: tutorials/standalone-package-based-installation.md [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+```
+# defguard-proxy -V
+defguard-proxy 0.5.0
+```
+
+## Running defguard
+
+### Run core
+
+To run basic core service we need to configure few env variables. We recommend to create a new env file where you can save [Database configuration](https://defguard.gitbook.io/defguard/admin-and-features/setting-up-your-instance/configuration#database-configuration) and few of [Core configuration](https://defguard.gitbook.io/defguard/admin-and-features/setting-up-your-instance/configuration#core-configuration).
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: @cpprian

1. this is a guide and it should not point to other guides/docs. It should explain everything.

2. there should not be ever configuration through ENV vars - that is for docker.
Packages should be configured by config files - which core/gateway support.

3. There should be a **complete example** of a config file. For example use testing server and it's domains and IP and show using those how it's configured.

File: tutorials/standalone-package-based-installation.md [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+* Debian-based operating system (Debian, Ubuntu, etc.)
+* Administrative (sudo) privileges
+* Internet connection for downloading packages
+
+### Prequesities
+
+#### PostgreSQL
+
+Defguard services utilitize postgresql so if you do not have installed and configured yet, you can do it with this [guide](https://wiki.debian.org/PostgreSql). For this tutorial we need to create **a user with superuser priviliges and database**. Remember database configuration, because we will need it during [Run Core](#run-core) section.
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: Missing commands for creating database and user.

File: tutorials/standalone-package-based-installation.md [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+
+### Run gateway
+
+To run gateway, we should do two things:
+* setup our first location on core service that you create earlier to get `token` and `grpc_url` variables,
+* create a new config.toml file. 
+
+#### Setup location for gateway
+
+Now, after setting up core service you should go to the website that you set on `DEFGUARD_URL` (if this variable was not set by you, go to page: `http://localhost:8000`). The link should redirect you to login page, if you not set your [own password](https://defguard.gitbook.io/defguard/admin-and-features/setting-up-your-instance/configuration#core-configuration) for admin the data should look like this:
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: Missing example with a domain - use the domain my-server.defguard.net and IP used for the testing server as example.

File: tutorials/standalone-package-based-installation.md [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+# A HTTP port that will expose the REST HTTP gateway health status
+# STATUS CODES:
+# 200 - Gateway is working and is connected to CORE
+# 503 - gateway works but is not connected to CORE
+#health_port = 55003
+```
+
+Now we can run gateway service with configuration above:
+```
+# defguard-gateway --config <path_to_your_config.toml>
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: Gateway is run by a systemd service not a command you invoke by hand

**Comment by teon** [comment]
@cpprian GRPC ports should absolutely not be exposed! They are internal ports for internal communication. Please remove that statement from guide.

File: tutorials/standalone-package-based-installation.md [hunk 5]
```diff
@@ -497,10 +499,176 @@ To run proxy service, we can do it by:
...
+DEFGUARD_ADMIN_GROUPNAME=admin
+DEFGUARD_DEFAULT_ADMIN_PASSWORD=pass123
+
+DEFGUARD_GRPC_ULR=https://my-server.defguard.net # add this line to your config file
+
+### Proxy configuration ###
+# Optional. URL of proxy gRPC server
+DEFGUARD_PROXY_URL=https://enroll.defguard.net
+
+### LDAP configuration ###
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md:635]: @cpprian this should be commended out and indicated that's only when someone wants sync to LDAP (which is not a part of this tutorial)

File: tutorials/standalone-package-based-installation.md [hunk 6]
```diff
@@ -497,10 +499,176 @@ To run proxy service, we can do it by:
...
+DEFGUARD_SECRET_KEY=9oZqdHRCN0TWIyMhjYOAYwgzVz9IfOqz62PzUvjvyMzqLICGSM3b0pRMdDH300CQ
+DEFGUARD_URL=https://my-server.defguard.net
+# How long auth session lives in seconds
+DEFGUARD_AUTH_SESSION_LIFETIME=604800
+# Optional. Generated based on DEFGUARD_URL if not provided.
+# DEFGUARD_WEBAUTHN_RP_ID=localhost
+DEFGUARD_ADMIN_GROUPNAME=admin
+DEFGUARD_DEFAULT_ADMIN_PASSWORD=pass123
+
+DEFGUARD_GRPC_ULR=https://my-server.defguard.net # add this line to your config file
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: @cpprian this is incorrect - the port is 444 for grpc so it should be: https://my-server.defuard.net:444

File: tutorials/standalone-package-based-installation.md [hunk 7]
```diff
@@ -0,0 +1,648 @@
...
+defguard=# exit     # for now we can leave it, the purpose of this connection is to verify your user is able to communicate with database
+```
+
+#### NGINX
+
+To expose our services in the server we need to configure a reverse proxy server. For this we will use nginx web server with ssl certificates for enabling https protocol.
+
+To get started, we need to install:
+```
+# apt install nginx
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: @cpprian why this way? when you could do this in one command: apt install nginx certbot?

File: tutorials/standalone-package-based-installation.md [hunk 8]
```diff
@@ -0,0 +1,648 @@
...
+# systemctl enable nginx.service
+# systemctl start nginx.service
+```
+
+### Additional dependencies
+
+If there are any missing dependencies during installation, we recommend to fix them using the following command:
+
+```
+# apt-get install -f
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: @cpprian this command indicates that something went wrong during the installation. If there are dependecies that defguard requires plese indicate that by: apt install dep1 dep2 dep3 (and not apt install -f)

File: tutorials/standalone-package-based-installation.md [hunk 9]
```diff
@@ -0,0 +1,648 @@
...
+# Optional. Generated based on DEFGUARD_URL if not provided.
+# DEFGUARD_WEBAUTHN_RP_ID=localhost
+DEFGUARD_ADMIN_GROUPNAME=admin
+DEFGUARD_DEFAULT_ADMIN_PASSWORD=pass123
+
+DEFGUARD_GRPC_ULR=https://my-server.defguard.net # add this line to your config file
+
+### Proxy configuration ###
+# Optional. URL of proxy gRPC server
+# DEFGUARD_PROXY_URL=http://localhost:50051
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md:193]: @cpprian this is missing... we are installing enrollment (it should be: https://enrollment.defguard.net:444)

File: tutorials/standalone-package-based-installation.md [hunk 10]
```diff
@@ -0,0 +1,648 @@
...
+```
+# This is an example config file for defguard VPN gateway
+# To use it fill in actual values for your deployment below
+
+# Required: secret token generated by defguard
+# NOTE: must replace default with actual value
+token = "<your_gateway_token>"
+# Required: defguard server gRPC endpoint URL
+# NOTE: must replace default with actual value
+grpc_url = "<defguard_grpc_url>"
```
Comment (teon) [inline tutorials/standalone-package-based-installation.md]: @cpprian this is missing - it should have an example according to this setup: https://my-server.defguard.net:444

File: tutorials/standalone-package-based-installation.md (hunk 9 above)
Comment (cpprian) [reply tutorials/standalone-package-based-installation.md:193]: `DEFGUARD_PROXY_URL` is not missing, it is set during the `Run proxy` section. You're right that I forgot to add port 444...

**Review by teon — APPROVED** [review]

**Pull request merged**